x-pack/plugins/security_solution/server/endpoint/routes/resolver/utils/ancestry_query_handler.ts

```diff
@@ -74,8 +74,26 @@ export class AncestryQueryHandler implements QueryHandler<ResolverAncestry> {
     // bucket the start and end events together for a single node
     const ancestryNodes = this.toMapOfNodes(results);
 
-    // the order of this array is going to be weird, it will look like this
-    // [furthest grandparent...closer grandparent, next recursive call furthest grandparent...closer grandparent]
+    /**
+     * the order of this array is going to be weird, it will look like this
+     * [furthest grandparent...closer grandparent, next recursive call furthest grandparent...closer grandparent]
+     *
+     * Consider the following tree:
+     * A -> B -> C -> D -> E -> Origin
+     * Where A was spawn before B, which was before C, etc
+     *
+     * Let's assume the ancestry array size is 2 so Origin's array would be: [E, D]
```

Comment:
:grey_question: Above, you assert :

> it will look like this
     * [furthest grandparent...closer grandparent, next recursive call furthest grandparent...closer grandparent]

But none of the examples you cite seem to satisfy that pattern (e.g E is not the furthest grandparent)... maybe saying they will be iterated in queue order batched in chunks of `ancestor size` or something...

Reply:
That statement is referring to the accumulated ancestors which is ultimately returned to the API caller. The examples I give here are referring to the ancestry array contents stored with each process document. I'll update the comment to make that more clear.

Origin's document in ES has `[E, D]` as the contents of the ancestry array. E's document in ES has `[D, C]`, etc.

The full accumulated results array is `[D, E, B, C, A]` which is in the order that the comment describes:

For iteration 1: `D` is the furthest grandparent from Origin and `E` is the closest. For iteration 2: `B` is the furthest grandparent from `D` etc.